src/pymor/discretizers/builtin/cg.py

```diff
@@ -1183,6 +1208,26 @@ def discretize_stationary_cg(analytical_problem, diameter=None, domain_discretiz
 
     data = {'grid': grid, 'boundary_info': boundary_info}
 
+    if energy_product:
+        if preassemble:
+            # energy_product is the |Parameter| with which we build the energy product (s.a.)
+            eL = eL.assemble(energy_product)
+        else:
+            from pymor.operators.constructions import FixedParameterOperator
+            eL = FixedParameterOperator(eL, mu=energy_product)
+        products = m.products.copy()
+        if p.diffusion is not None:
+            scalar_diffusion = len(p.diffusion.shape_range) == 0 \
```

Comment:
Is that first case even allowed?

Reply:
Why shouldn't it be allowed? It corresponds to shape_range == ()

Reply:
Sorry, my bad.